gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/service/RedirectUriValidator.java

```diff
@@ -49,4 +58,8 @@ public void validate(Client client, String requestedRedirectUri, BiConsumer<Stri
             checkMethod.accept(requestedRedirectUri, registeredClientRedirectUris);
         }
     }
+
+    private boolean requiresRedirectUri(TokenPurpose operation) {
+        return operation != TokenPurpose.RESET_PASSWORD;
```

Comment:
**suggestion**: could be nice to specify in a comment the reason why there is an exception for reset_password.